src/components/OwnerHome/index.js

```diff
@@ -41,6 +46,49 @@ export default class OwnerHome extends Component {
     this.setState({ addSlots: !this.state.addSlots });
   };
 
+  componentDidMount() {
+    const { user } = this.state;
+    const userId =
+      loadState('userAuthenticationDetails') &&
+      loadState('userAuthenticationDetails').userId;
+      
+    axios
+      .get(
+        `https://safeslot-backend.herokuapp.com/api/users/${userId}?filter={"include": "stores"}`
+      )
+      .then(res => {
+        const user = res.data;
+        this.setState({ user });
+        if (!user.isStoreOwner) {
+          this.props.history.push('/');
+          return;
+        }
+        if (
+          (user.isStoreOwner && !user.storeId) ||
+          !user.stores ||
+          Object.keys(user.stores).length === 0
```

Comment:
super small - you can also do: `!Object.keys(user.stores).length` because if the length is 0, it's falsey.